test/e2e/reconciler/scheduling/placement_scheduler_test.go

```diff
@@ -0,0 +1,341 @@
+	framework.Eventually(t, func() (bool, string) {
+		placement, err := kcpClusterClient.Cluster(userClusterName).SchedulingV1alpha1().Placements().Get(ctx, "default", metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Sprintf("Failed to get placement: %v", err)
+		}
+
+		return placement.Status.Phase == schedulingv1alpha1.PlacementPending, ""
+	}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+	t.Logf("Create a new placement to include the location")
+	newPlacement := &schedulingv1alpha1.Placement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "new-placement",
+		},
+		Spec: schedulingv1alpha1.PlacementSpec{
+			LocationSelectors: []metav1.LabelSelector{{}},
+			NamespaceSelector: &metav1.LabelSelector{},
+			LocationResource: schedulingv1alpha1.GroupVersionResource{
+				Group:    "workload.kcp.dev",
+				Version:  "v1alpha1",
+				Resource: "workloadclusters",
+			},
+			LocationWorkspace: locationClusterName.String(),
+		},
+	}
+	_, err = kcpClusterClient.Cluster(userClusterName).SchedulingV1alpha1().Placements().Create(ctx, newPlacement, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Wait for new placement to be ready")
+	framework.Eventually(t, func() (bool, string) {
+		placement, err := kcpClusterClient.Cluster(userClusterName).SchedulingV1alpha1().Placements().Get(ctx, newPlacement.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Sprintf("Failed to get placement: %v", err)
+		}
+
+		return conditions.IsTrue(placement, schedulingv1alpha1.PlacementReady), fmt.Sprintf("placement is not ready: %s", toYaml(binding))
+	}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+	t.Logf("Wait for resource to by synced again")
+	framework.Eventually(t, func() (bool, string) {
+		svc, err := kubeClusterClient.Cluster(userClusterName).CoreV1().Services("default").Get(ctx, "first", metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Sprintf("Failed to get service: %v", err)
+		}
+
+		if len(svc.Annotations[workloadv1alpha1.InternalClusterDeletionTimestampAnnotationPrefix+firstWorkloadClusterName]) != 0 {
+			return false, fmt.Sprintf("resource should not be removed but got %s", toYaml(svc))
+		}
+		return svc.Labels[workloadv1alpha1.InternalClusterResourceStateLabelPrefix+firstWorkloadClusterName] == string(workloadv1alpha1.ResourceStateSync), ""
+	}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+	t.Logf("Wait for the service to be sync to the downstream cluster")
+	require.Eventually(t, func() bool {
+		downstreamServices, err = syncerFixture.DownstreamKubeClient.CoreV1().Services("").List(ctx, metav1.ListOptions{
+			LabelSelector: "test.workload.kcp.dev=" + firstWorkloadClusterName,
+		})
+		if errors.IsNotFound(err) {
+			return false
+		} else if err != nil {
+			klog.Errorf("Failed to list Services: %v", err)
+			return false
+		} else if len(downstreamServices.Items) < 1 {
+			return false
+		}
+		return true
+	}, wait.ForeverTestTimeout, time.Millisecond*100)
+}
```

Comment:
this needs another case where both placements against different locations exist and we sync twice.

Unrelated or not: wondering, what do we do if two placements select the same location? Or if two clusters belong to the same location? I.e. all these cases of overlap.